### PR TITLE
fix: deploy errors based on subscription status

### PR DESCRIPTION
**Pull request by AdityaHegde**
Updates the deploy error page to be more descriptive based on subscription status. More details: https://www.notion.so/rilldata/Rill-Developer-deploy-trial-information-modal-cases-135ba33c8f5780f6959ae5873e7c44cb?d=13cba33c8f5780eabf04001c47642617#138ba33c8f5780b5b740c20fdca6b422

- [x] Skip showing the "Start trial" confirm dialog during deploy if the user has any org with a plan attached.
- [x] Show the "Start trial" confirm dialog when the selected org during deploy is a blank org without a plan attached.
- [x] Parse deploy errors and show upgrade button when quota is hit or subscription has ended.
- [x] The upgrade button from the above should take the user to rill cloud's billing page with upgrade modal open.
- [x] Upgrading successfully in clould should redirect user back to deploy. Note that if the user navigates to some other page here before upgrading then the redirect will no longer apply.

### CODE REVIEW

File: proto/rill/local/v1/api.proto [hunk 1]
```diff
@@ -28,6 +28,9 @@ service LocalService {
 
   // GetCurrentProject returns the rill cloud project connected to the local project
   rpc GetCurrentProject(GetCurrentProjectRequest) returns (GetCurrentProjectResponse) {}
+
+  // GetUserOrgMetadata returns metadata about the current user's orgs.
+  rpc GetUserOrgMetadata(GetUserOrgMetadataRequest) returns (GetUserOrgMetadataResponse) {}
```
Comment (pjain1) [inline proto/rill/local/v1/api.proto:33]: may be call it `ListOrganizationsBillingIssues` ?

File: proto/rill/local/v1/api.proto [hunk 2]
```diff
@@ -108,3 +111,15 @@ message GetCurrentProjectResponse {
   string local_project_name = 1;
   admin.v1.Project project = 2;
 }
+
+message GetUserOrgMetadataRequest {}
+
+message GetUserOrgMetadataResponse {
+  message OrgMetadata {
+    string name = 1;
+    bool is_admin = 2;
```
Comment (pjain1) [inline proto/rill/local/v1/api.proto:120]: this is not populated in the API ? is this used ?

Comment (AdityaHegde) [reply proto/rill/local/v1/api.proto:120]: No I removed it in further iterations.

File: web-admin/src/features/navigation/getSingleUseUrlParam.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      // no-op
+    }
+    const redirectUrl = new URL(url);
+    redirectUrl.searchParams.delete(param);
+    if (url.searchParams.has("redirect")) {
+      redirectUrl.searchParams.set(
+        "redirect",
+        url.searchParams.get("redirect"),
+      );
+    }
```
Comment (ericpgreen2) [inline web-admin/src/features/navigation/getSingleUseUrlParam.ts:23]: Shouldn't we retain all other search parameters (besides the transient `param`), not just `redirect`?

**Review by ericpgreen2 — CHANGES_REQUESTED** [review]

File: web-admin/src/features/navigation/getSingleUseUrlParam.ts (hunk 3 above)
Comment (AdityaHegde) [reply web-admin/src/features/navigation/getSingleUseUrlParam.ts:23]: Doing `new URL(url)` achieves that.

File: web-common/src/features/dashboards/workspace/DeployProjectCTA.svelte [hunk 4]
```diff
@@ -69,6 +66,13 @@
   }
 
   function onShowDeploy() {
+    if (!isFirstTimeDeploy) {
+      // do not show the confirmation dialog for successive deploys
+      void behaviourEvent?.fireDeployEvent(BehaviourEventAction.DeployIntent);
+      window.open(deployCTAUrl, "_target");
```
[the remark this answers is not in the document]
Comment (AdityaHegde) [reply web-common/src/features/dashboards/workspace/DeployProjectCTA.svelte:72]: Ya it is supposed to be `_blank`

File: web-local/src/routes/(misc)/deploy/+page.svelte [hunk 5]
```diff
@@ -1,43 +1,73 @@
...
   }
 
   onMount(() => {
-    void deployer.loginOrDeploy();
+    if (orgParam) {
+      // org param is sent as a callback after starting a team plan.
+      // so directly deploy using that org
+      void deployer.deploy(orgParam);
+    } else {
+      void deployer.loginOrDeploy();
```
[the remark this answers is not in the document]
Comment (AdityaHegde) [reply web-local/src/routes/(misc)/deploy/+page.svelte:68]: I think adding user login makes sense here. But adding org choosing would be over kill and just moves complexity to the other parts IMO.

[the remark this answers is not in the document]
Comment (AdityaHegde) [reply web-local/src/routes/(misc)/deploy/+page.svelte]: I refactored to not need this. Moved the orgParam to deployer's constructor. Feels cleaner

File: web-local/src/routes/(misc)/deploy/+page.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+} from "@rilldata/web-common/runtime-client/local-service";
+
+export const load = async ({ url }) => {
+  const orgMetadata = await queryClient.fetchQuery({
+    queryKey: getLocalServiceGetUserOrgMetadataRequestQueryKey(),
+    queryFn: () => localServiceGetUserOrgMetadataRequest(),
+    staleTime: Infinity,
+  });
+  return {
+    orgParam: url.searchParams.get("org") ?? "",
```
[the remark this answers is not in the document]
Comment (AdityaHegde) [reply web-local/src/routes/(misc)/deploy/+page.ts:15]: Using this instead of getting it from params in `+page.svelte`

**Review by pjain1 — APPROVED** [review]
Good from backend pov

File: web-local/src/routes/(misc)/deploy/+page.svelte [hunk 7]
```diff
@@ -1,43 +1,64 @@
...
+      return;
+    }
+
     promptOrgSelection.set(false);
     return deployer.deploy(org);
   }
 
-  function onContactUs() {
-    window.Pylon("show");
+  function onOrgConfirm() {
```
Comment (ericpgreen2) [inline web-local/src/routes/(misc)/deploy/+page.svelte:54]: `onConfirmOrgTrial`?

**Review by ericpgreen2 — APPROVED** [review]
Code looks good. I just haven't QA'd each of these permutations, so I'll approve assuming you have.

**Comment by AdityaHegde** [comment]
Thanks @ericpgreen2 . We did a round of QA with @ericokuma and @jkhwu

**Pull request merged**